cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java

```diff
@@ -91,7 +91,9 @@ protected final void setReducerResources(Resources resources) {
 
   /**
    * Specifies set of Dataset names that are used by the {@link MapReduce}.
+   * @deprecated datasets used in runtime need not be specified during configuration
    */
+  @Deprecated
   protected final void useDatasets(String dataset, String...moreDatasets) {
     List<String> datasets = new ArrayList<String>();
     datasets.add(dataset);
```

Comment:
We have marked it as deprecated this release. So if the user is still using useDatasets, he can get the info about MR, Workers associated with the dataset. We can remove this logic and the method completely in 3.0.